### PR TITLE
Unit testing for GTID/GtidSet

### PR DESCRIPTION
**Pull request by darnuria**
It's a bit redundant with test about GtidSet but if it makes sense I can add more.

### CODE REVIEW

**Comment by darnuria** [comment]
Had some fun doing in rust the whole GTID thing so I wil contribute back the unit testing + doc. :)

**Comment by darnuria** [comment]
I think this one is ready for review. I may have left a little bit too much of redundancy in test about add/sub but it will not hurt I suppose.

File: pymysqlreplication/tests/test_basic.py [hunk 1]
```diff
@@ -902,6 +902,74 @@ def test_gtidset_representation_payload(self):
...
+    def test_parsing(self):
+        with self.assertRaises(ValueError) as exc:
+            gtid = Gtid("57b70f4e-20d3-11e5-a393-4a63946f7eac:1-5 57b70f4e-20d3-11e5-a393-4a63946f7eac:1-56")
+            gtid = Gtid("NNNNNNNN-20d3-11e5-a393-4a63946f7eac:1-5")
+            gtid = Gtid("-20d3-11e5-a393-4a63946f7eac:1-5")
+            gtid = Gtid("-20d3-11e5-a393-4a63946f7eac:1-")
+            gtid = Gtid("57b70f4e-20d3-11e5-a393-4a63946f7eac:A-1")
+            gtid = Gtid("57b70f4e-20d3-11e5-a393-4a63946f7eac:-1")
+            gtid = Gtid("57b70f4e-20d3-11e5-a393-4a63946f7eac:1-:1")
+            gtid = Gtid("57b70f4e-20d3-11e5-a393-4a63946f7eac::1")
```
Comment (darnuria) [inline pymysqlreplication/tests/test_basic.py:970]: Leaved the local variable for debugging purpose but I can remove it.

**Comment by julien-duponchelle** [comment]
Thanks !

**Comment by darnuria** [comment]
Thank for the review time!

**Pull request merged**